### PR TITLE
expand dependency

### PR DESCRIPTION
**Pull request by ansyral**

### CODE REVIEW

**Comment by hellosnow** [comment]
:shipit:

**Comment by ansyral** [comment]
#449

File: src/Microsoft.DocAsCode.Build.Engine/Incrementals/ChangeList.cs [hunk 1]
```diff
@@ -12,7 +13,7 @@ namespace Microsoft.DocAsCode.Build.Engine.Incrementals
     using Microsoft.DocAsCode.Exceptions;
     using Microsoft.DocAsCode.Utility;
 
-    public sealed class ChangeList
+    public sealed class ChangeList : IEnumerable
```
Comment (vwxyzh) [inline src/Microsoft.DocAsCode.Build.Engine/Incrementals/ChangeList.cs]: implements IEnumerable<ChangeItem> ?

File: src/Microsoft.DocAsCode.Build.Engine/SingleDocumentBuilder.cs [hunk 2]
```diff
@@ -198,9 +198,40 @@ private bool GetCanIncremental(string versionName)
...
+        private void ExpandDependency(DocumentBuildParameters parameter, BuildInfo lastBuildInfo, Dictionary<string, ChangeKindWithDependency> changeItems)
         {
-            // todo ExpandDependency;
+            string versionName = parameter.VersionName;
+            string dependencyFile = lastBuildInfo.Versions.SingleOrDefault(v => v.VersionName == versionName)?.Dependency;
+
+            if (changeItems == null)
+            {
+                changeItems = new Dictionary<string, ChangeKindWithDependency>();
+            }
```
Comment (superyyrrzz) [inline src/Microsoft.DocAsCode.Build.Engine/SingleDocumentBuilder.cs]: move this outside ExpandDependency()? The dict created here can't be passed to outside context.ChangeDict.

**Pull request merged**